src/constants.js

```diff
@@ -28,3 +31,8 @@ export const statusFilter = [
   { label: 'ui-users.filters.status.active', value: 'active' },
   { label: 'ui-users.filters.status.inactive', value: 'inactive' }
 ];
+
+export const renderPermission = (name) => {
+  const [pPrefix, ...pName] = name.split('.');
+  return <FormattedMessage id={`${pPrefix}.permission.${pName.join('.')}`} />;
+};
```

Comment:
At least in the short term, when almost all modules will not have these translations, I think we need to fall back to the `displayName` in the absence of a translation. That's may be hard to do with `<FormattedMessage>` but is probably relatively easy with `intl.formatMessage`.

Reply:
Also, I don't understand why this function is in a file called `constants.js`. Surely something like `src/util/renderPermission.js` would make more sense?

Reply:
You are correct on all fronts. My answer on all fronts is, "I am lazy." I totally recognize that I cherry-picked the easy parts and left the thorny bits un-handled. 

`constants` was there so I used it; sorting looked painful so I skipped it. Maybe there's enough here to get started with for somebody else whose team is actually allocated to this task to finish it off?

Reply:
Oh, it's easy to do this with `<FormattedMessage>` after all, using the `defaultMessage` prop:
```
<FormattedMessage id={`${pPrefix}.permission.${pName.join('.')}`} defaultMessage={...}/>;
};
```

Reply:
I think you're better than that :-)